### PR TITLE
Use partition projection when creating the CloudTrail Athena table

### PR DESCRIPTION
**Pull request by dimaryaz**
Saves us a whole bunch of `ALTER TABLE` calls - and hopefully fixes the lambda.

### CODE REVIEW

**Comment by akarve** [comment]
btw if you end up storing a stride or any other arguments in `LAST_UPDATE_KEY` then we should probably just make it JSON in case we need to store/reconstruct state with json.dumps/loads.

File: lambdas/access_counts/index.py [hunk 1]
```diff
@@ -85,20 +85,26 @@ def sql_escape(s):
...
+    LOCATION 's3://{bucket}/AWSLogs/'
+    TBLPROPERTIES (
+        'classification' = 'cloudtrail',
+        'projection.enabled' = 'true',
+        'projection.account.type' = 'enum',
+        'projection.account.values' = '{accounts}',
+        'projection.region.type' = 'enum',
+        'projection.region.values' = '{regions}',
+        'projection.date.type' = 'date',
+        'projection.date.range' = '{start_date},{end_date}',
```
[the remark this answers is not in the document]
Comment (dimaryaz) [reply lambdas/access_counts/index.py:101]: I've thought about it... Needs to be `NOW-1YEAR` or something in order for the backfill to work. Not sure if there's a cost to add more partitions than needed; maybe it's negligible. I just figured, we already have the exact dates - might as well use them.

File: lambdas/access_counts/index.py [hunk 2]
```diff
@@ -85,20 +85,26 @@ def sql_escape(s):
...
+        'projection.date.format' = 'yyyy/MM/dd',
+        'projection.date.interval' = '1',
```
[the remark this answers is not in the document]
Comment (dimaryaz) [reply lambdas/access_counts/index.py:103]: Sure

File: lambdas/access_counts/index.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (akarve) [reply lambdas/access_counts/index.py:101]: Agree with your point about timestamps.

**Comment by dimaryaz** [comment]
Regarding stride: need to think about that. Some transient failures could cut the stride needlessly - and it'll stay that way unless we have logic to increase it again.

A better solution might be to process some small number of days at a time in a loop, within a single lambda invocation. Need some logic to check how much time is remaining.

One annoying problem: if the lambda goes over the time limit, `object_access_log` (probably) ends up with partial data. And it does not support `DELETE` - but I guess we can manually delete directories in S3.

In any case: it's probably worth doing something here, but I think it should be a separate change.

**Comment by akarve** [comment]
OK, let's take this change and test it out. Your loop idea is way better. Can you file to yourself? There is a function to get time remaining, from the `context` [here](https://github.com/quiltdata/quilt/blob/master/lambdas/es/indexer/document_queue.py#L196).

**Comment by akarve** [comment]
And in the new smarter version, we shouldn't have to delete anything that's older than 24 hours, because that should never change?

**Pull request merged**